commit 31e83e21cf00fe5b669eb352ff3ed70e74b40fad
Author: Paolo Bonzini <pbonzini@redhat.com>
Date:   Thu Sep 29 13:20:14 2022 -0400

    KVM: x86: compile out vendor-specific code if SMM is disabled
    
    Vendor-specific code that deals with SMI injection and saving/restoring
    SMM state is not needed if CONFIG_KVM_SMM is disabled, so remove the
    four callbacks smi_allowed, enter_smm, leave_smm and enable_smi_window.
    The users in svm/nested.c and x86.c also have to be compiled out; the
    amount of #ifdef'ed code is small and it's not worth moving it to
    smm.c.
    
    enter_smm is now used only within #ifdef CONFIG_KVM_SMM, and the stub
    can therefore be removed.
    
    Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
    Reviewed-by: Maxim Levitsky <mlevitsk@redhat.com>
    Message-Id: <20220929172016.319443-7-pbonzini@redhat.com>
    Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>

diff --git a/arch/x86/include/asm/kvm_host.h b/arch/x86/include/asm/kvm_host.h
index 612ef60631c1..3e5e54d7baa6 100644
--- a/arch/x86/include/asm/kvm_host.h
+++ b/arch/x86/include/asm/kvm_host.h
@@ -1612,10 +1612,12 @@ struct kvm_x86_ops {
 
 	void (*setup_mce)(struct kvm_vcpu *vcpu);
 
+#ifdef CONFIG_KVM_SMM
 	int (*smi_allowed)(struct kvm_vcpu *vcpu, bool for_injection);
 	int (*enter_smm)(struct kvm_vcpu *vcpu, char *smstate);
 	int (*leave_smm)(struct kvm_vcpu *vcpu, const char *smstate);
 	void (*enable_smi_window)(struct kvm_vcpu *vcpu);
+#endif
 
 	int (*mem_enc_ioctl)(struct kvm *kvm, void __user *argp);
 	int (*mem_enc_register_region)(struct kvm *kvm, struct kvm_enc_region *argp);